src/Microsoft.IdentityModel.Tokens/TokenHandler.cs

```diff
@@ -73,5 +74,26 @@ public int TokenLifetimeInMinutes
             get => _defaultTokenLifetimeInMinutes;
             set => _defaultTokenLifetimeInMinutes = (value < 1) ? throw LogExceptionMessage(new ArgumentOutOfRangeException(nameof(value), FormatInvariant(LogMessages.IDX10104, LogHelper.MarkAsNonPII(value)))) : value;
         }
+
+        #region methods
+
+        /// <summary>
+        /// Validates a token.
+        /// </summary>
+        /// <param name="token">The token to be validated.</param>
+        /// <param name="validationParameters">A <see cref="TokenValidationParameters"/> required for validation.</param>
+        /// <returns>A <see cref="TokenValidationResult"/></returns>
+        public virtual Task<TokenValidationResult> ValidateTokenAsync(string token, TokenValidationParameters validationParameters) => throw new NotImplementedException();
+
+        /// <summary>
+        /// Converts a string into an instance of <see cref="SecurityToken"/>.
+        /// </summary>
+        /// <param name="token">The string to be deserialized.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="token"/> is null or empty.</exception>
+        /// <exception cref="ArgumentException">'token.Length' is greater than <see cref="TokenHandler.MaximumTokenSizeInBytes"/>.</exception>
+        /// <returns>A <see cref="SecurityToken"/>.</returns>
+        public virtual SecurityToken ReadToken(string token) => throw new NotImplementedException();
```

Comment:
>ReadToken [](http://example.com/codeflow?start=37&length=9)

How do we think about adding methods that will throw be default in the "Breaking changes" paradigm? Obviously we have to give it some default behavior if we're adding something to the contract here and throwing an exception makes the most sense but do we face an issue if someone had 

A : TokenHandler (for the old version)

and now when they pass A in, it throws an exception in cases where it perhaps wouldn't have before? Or if someone inherits from JsonWebTokenHandler wouldn't this virtual method be the default now instead of the now override method in that class for ReadToken? Isn't that breaking? I'm probably missing something but would appreciate a run down :)

Reply:
1. If someone has A : TokenHandler, there should not be any way for their existing code to call this method since this is new. Do you have any specific scenario in mind where this will throw an exception when it would not before?

2. If someone inherits from JsonWebTokenHandler and they don't provide their own implementation of ReadToken(), then the implementation from JsonWebTokenHandler  will be used so no exception will be thrown.

Reply:
I think you're right here, I'd gotten mixed up on how the inheritance would work.